frontend/src/container/BillingContainer/BillingUsageGraph/BillingUsageGraph.tsx

```diff
@@ -0,0 +1,171 @@
+import './BillingUsageGraph.styles.scss';
+import '../../../lib/uPlotLib/uPlotLib.styles.scss';
+
+import { Color } from '@signozhq/design-tokens';
+import { Card, Flex, Typography } from 'antd';
+import { getComponentForPanelType } from 'constants/panelTypes';
+import { PANEL_TYPES } from 'constants/queryBuilder';
+import { PropsTypePropsMap } from 'container/GridPanelSwitch/types';
+import { useIsDarkMode } from 'hooks/useDarkMode';
+import { useResizeObserver } from 'hooks/useDimensions';
+import tooltipPlugin from 'lib/uPlotLib/plugins/tooltipPlugin';
+import getAxes from 'lib/uPlotLib/utils/getAxes';
+import getRenderer from 'lib/uPlotLib/utils/getRenderer';
+import { getUPlotChartData } from 'lib/uPlotLib/utils/getUplotChartData';
+import { getXAxisScale } from 'lib/uPlotLib/utils/getXAxisScale';
+import { getYAxisScale } from 'lib/uPlotLib/utils/getYAxisScale';
+import { FC, useRef } from 'react';
+import uPlot from 'uplot';
+
+import {
+	convertDataToMetricRangePayload,
+	fillMissingValuesForQuantities,
+} from './utils';
+
+interface BillingUsageGraphProps {
+	data: any;
+	billAmount: number;
+}
+const paths = (
+	u: any,
+	seriesIdx: number,
+	idx0: number,
+	idx1: number,
+	extendGap: boolean,
+	buildClip: boolean,
+): uPlot.Series.PathBuilder => {
+	const s = u.series[seriesIdx];
+	const style = s.drawStyle;
+	const interp = s.lineInterpolation;
+
+	const renderer = getRenderer(style, interp);
+
+	return renderer(u, seriesIdx, idx0, idx1, extendGap, buildClip);
+};
+
+export function BillingUsageGraph(props: BillingUsageGraphProps): JSX.Element {
+	const { data, billAmount } = props;
+	const graphCompatibleData = convertDataToMetricRangePayload(data);
+	const chartData = getUPlotChartData(graphCompatibleData);
+	const graphRef = useRef<HTMLDivElement>(null);
+	const isDarkMode = useIsDarkMode();
+	const containerDimensions = useResizeObserver(graphRef);
+
+	const { billingPeriodStart: startTime, billingPeriodEnd: endTime } = data;
+
+	const Component = getComponentForPanelType(PANEL_TYPES.BAR) as FC<
+		PropsTypePropsMap[PANEL_TYPES]
+	>;
+
+	const getGraphSeries = (color: string, label: string): any => ({
+		drawStyle: 'bars',
+		paths,
+		lineInterpolation: 'spline',
+		show: true,
+		label,
+		fill: color,
+		stroke: color,
+		width: 2,
+		spanGaps: true,
+		points: {
+			size: 5,
+			show: false,
+			stroke: color,
+		},
+	});
+
+	const uPlotSeries: any = [
+		{ label: 'Timestamp', stroke: 'purple' },
+		getGraphSeries(
+			'#DECCBC',
+			graphCompatibleData.data.result[0]?.legend as string,
+		),
+		getGraphSeries(
+			'#4E74F8',
+			graphCompatibleData.data.result[1]?.legend as string,
+		),
+		getGraphSeries(
+			'#F24769',
+			graphCompatibleData.data.result[2]?.legend as string,
+		),
+	];
+
+	const axesOptions = getAxes(isDarkMode, '');
+
+	const getOptionsForChart: uPlot.Options = {
+		id: 'billing-usage-breakdown',
+		series: uPlotSeries,
```

Comment:
can we memoize this ? as this will be recalculated on every re-render

Reply:
done